src/v8n.js

```diff
@@ -40,16 +40,16 @@ function proxyContext(context) {
 }
 
 function proxyContextEs5(context) {
-  const buildRuleSet = ruleSet =>
+  const addRuleSet = (ruleSet, targetContext) =>
     Object.keys(ruleSet).forEach(prop => {
-      context[prop] = (...args) => {
-        const newContext = proxyContextEs5(context._clone());
+      targetContext[prop] = (...args) => {
+        const newContext = proxyContextEs5(targetContext._clone());
         return newContext._applyRule(ruleSet[prop], prop)(...args);
       };
     });
 
-  buildRuleSet(availableRules);
-  buildRuleSet(customRules);
+  addRuleSet(availableRules, context);
```

Comment:
The returned, changed context should be reassigned here. otherwise the context variable is changed below and it's not clear where. So it should be clearly reassigned.